policy.go

```diff
@@ -216,6 +228,76 @@ func (p *policyPlugin) parsePDP(c *caddy.Controller) error {
 	return nil
 }
 
+func (p *policyPlugin) evaluateExp(args []string, keyword string, w dns.ResponseWriter, r *dns.Msg) (interface{}, error) {
+	req := request.Request{W: w, Req: r}
+	parameters := make(map[string]interface{}, 8)
+
+	switch args[0] {
+	case "queryName":
+		args[0] = req.QName()
+		parameters["queryName"] = args[0]
+	}
+	expression, err := govaluate.NewEvaluableExpression(strings.Join(args, " "))
+	if err != nil {
+		return nil, err
+	}
+
+	result, err := expression.Evaluate(parameters)
+	if err != nil {
+		return nil, err
+	}
+	return result, nil
+}
+
+func (p *policyPlugin) evaluateExpression(args []string) (interface{}, error) {
+	expression, err := govaluate.NewEvaluableExpression(strings.Join(args, " "))
+	if err != nil {
+		return nil, err
+	}
+	parameters := make(map[string]interface{}, 8)
+	parameters["queryName"] = args[0]
+	result, err := expression.Evaluate(parameters)
+	if err != nil {
```

Comment:
yes with several points:

1- parameters will be extracted from context (and you need Metadata)
2- you need to verify you have a boolean as a result. Or ensure to make the result a boolean